vault/core.go

```diff
@@ -696,6 +696,9 @@ type Core struct {
 	// if populated, the callback is called for every request
 	// for testing purposes
 	requestResponseCallback func(logical.Backend, *logical.Request, *logical.Response)
+
+	// If role quotas are not enabled, don't track lease counts by role
```

Comment:
```suggestion
	// If no role-based lease-count quota is enabled, don't track lease counts by role
```

Reply:
I think this is just _any_ quota, with the current implementation. It only *affects* lease-count quotas, but if any role-based quota is enabled, we'll already have the role and thus account for it in the `leaseEntry`.

Reply:
I see why this naming is confusing. My understanding is that setting `impreciseLeaseRoleTracking=true` causes vault to skip "role lease tracking"* if and only if both RLQs and LCQs are not role based. 

*Role lease tracking = tracking the role of each lease even if a role based LCQ is not enabled.

Reply:
I changed this comment to read: `If no role-based lease count quota or role based rate limiting quota is enabled, don't track lease counts by role`

Reply:
Sounds good!